ravenframework/CodeInterfaceClasses/MELCOR/melcorCombinedInterface.py

```diff
@@ -13,104 +13,176 @@
-      @ In, fargs, dict, optional, a dictionary containing the axuiliary input file variables the user can specify in the input (e.g. under the node < Code >< clargstype =0 input0arg =0 aux0extension =0 .aux0/ >< /Code >)
+      @ In, fargs, dict, optional, a dictionary containing the auxiliary input file variables the user can specify in the input (e.g. under the node < Code >< fileargstype =0 input0arg =0 aux0extension =0 .aux0/ >< /Code >)
       @ In, preExec, string, optional, a string the command that needs to be pre-executed before the actual command here defined
       @ Out, returnCommand, tuple, tuple containing the generated command. returnCommand[0] is the command to run the code (string), returnCommand[1] is the name of the output root
     """
-    if preExec is None:
-      raise IOError('No preExec listed in input! Exiting...')
-    melcin,melgin = self.findInps(inputFiles)
-    #get the melgen part
-    melgCommand,melgOut = self.melcorInterface.generateCommand([melgin],preExec,clargs,fargs)
-    #get the melcor part
-    melcCommand,melcOut = self.melgenInterface.generateCommand([melcin],executable,clargs,fargs)
-    #combine them
-    returnCommand = melgCommand + melcCommand, melcOut
+    found = False
+
+    melcin = self.findInps(inputFiles)
+    if clargs:
+      precommand = executable + clargs['text']
+    else:
+      precommand = executable
+    melgCommand = str(preExec)    + ' ' + melcin.getFilename()
+    melcCommand = str(precommand) + ' ' + melcin.getFilename()
+    returnCommand = [('serial',melgCommand + ' && ' + melcCommand +' ow=o ')],self.melcOut
+
     return returnCommand
 
   def createNewInput(self,currentInputFiles,origInputFiles,samplerType,**Kwargs):
     """
-      This generates a new input file depending on which sampler has been chosen
+      This generates a new input file depending on which sampler is chosen
       @ In, currentInputFiles, list,  list of current input files (input files from last this method call)
       @ In, oriInputFiles, list, list of the original input files
       @ In, samplerType, string, Sampler type (e.g. MonteCarlo, Adaptive, etc. see manual Samplers section)
       @ In, Kwargs, dictionary, kwarded dictionary of parameters. In this dictionary there is another dictionary called "SampledVars"
              where RAVEN stores the variables that got sampled (e.g. Kwargs['SampledVars'] => {'var1':10,'var2':40})
       @ Out, newInputFiles, list, list of newer input files, list of the new input files (modified and not)
     """
-    return self.melcorInterface.createNewInput(currentInputFiles,origInputFiles,samplerType,**Kwargs)
 
-  def finalizeCodeOutput(self, command, output, workingDir):
+    if "dynamicevent" in samplerType.lower():
+      raise IOError("Dynamic Event Tree-based samplers not implemented for MELCOR yet! But we are working on that.")
+    indexes  = []
```

Comment:
It seems variable "indexes" is not used.

Reply:
removed